myhpom/models/user.py

```diff
@@ -1,8 +1,22 @@
 from django.db import models
 from django.contrib.auth.models import User
 from myhpom import validators
+import base64, hashlib
 
-"""Connect a pre_save receiver/hook to the User model, and use this model in all imports"""
+"""
+* Connect a pre_save receiver/hook to the User model, and use this model in all imports
+* monkeypatch User.__unicode__ to return the email address
+"""
+
+User.__unicode__ = lambda u: u.email
+
+def get_username(email):
+    """unique username as hash of the email. 
+    Must be <= 30 characters long. 30, Django? Really?
```

Comment:
django 2.1 source shows they bumped it up to 150. But in antiquated 1.8: really.